pkg/distributor/query.go

```diff
@@ -295,12 +349,20 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 				hashToTimeSeries[key] = existing
 			}
 		}
+
+		// Start buffering chunks for streaming series
+		if res.streamingSeries.StreamReader != nil {
+			res.streamingSeries.StreamReader.StartBuffering()
+		}
 	}
 
+	expectedZonesWithResults := replicationSet.ZoneCount() - replicationSet.MaxUnavailableZones
```

Comment:
A couple of comments here:
1. It assumes zone-aware replication is enabled, but be not
2. Why counting the zones at all? You want to estimate how many sources you will have for 1 series. Isn't the quorum what you want here (which is unrelated from zone-aware replication)?

Reply:
I think I've addressed this in f3e057c - could you please take another look?